examples/sampler.cpp

```diff
@@ -240,137 +255,156 @@ int main(int argc, char* argv[])
-    serialize["total-fine-iterations"] = picojson::value((double) total_fine_iterations);
+    serialize["total-coarse-iterations"] = picojson::value((double) total_iterations[1]);
+    serialize["total-fine-iterations"] = picojson::value((double) total_iterations[0]);
     serialize["fine-mean-typical"] = picojson::value(
-                                         mean_fine[mean_fine.Size() / 2]);
+                                         mean[0][mean[0].Size() / 2]);
     serialize["fine-mean-l1"] = picojson::value(
-                                    mean_fine.Norml1() / static_cast<double>(mean_fine.Size()));
+                                    mean[0].Norml1() / static_cast<double>(mean[0].Size()));
     serialize["coarse-mean-l1"] = picojson::value(
-                                      mean_upscaled.Norml1() / static_cast<double>(mean_upscaled.Size()));
+                                      mean[1].Norml1() / static_cast<double>(mean[1].Size()));
     serialize["coarse-mean-typical"] = picojson::value(
-                                           mean_upscaled[mean_upscaled.Size() / 2]);
+                                           mean[1][mean[1].Size() / 2]);
     serialize["fine-variance-mean"] = picojson::value(
-                                          m2_fine.Sum() / static_cast<double>(m2_fine.Size()));
+                                          m2[0].Sum() / static_cast<double>(m2[0].Size()));
     serialize["coarse-variance-mean"] = picojson::value(
-                                            m2_upscaled.Sum() / static_cast<double>(m2_upscaled.Size()));
+                                            m2[1].Sum() / static_cast<double>(m2[1].Size()));
     serialize["max-p-error"] = picojson::value(max_p_error);
+    for (int i = 0; i < num_levels; ++i)
+    {
+        std::stringstream s;
+        s << "p-error-level-" << i;
+        serialize[s.str()] = picojson::value(p_error[i]);
+    }
 
     if (visualization)
     {
-        Visualize(mean_upscaled, ufespace, 1);
-        Visualize(mean_fine, ufespace, 0);
-        if (count > 1.1)
+        for (int level = 0; level < num_levels; ++level)
         {
-            Visualize(m2_upscaled, ufespace, 11);
-            Visualize(m2_fine, ufespace, 10);
+            Visualize(mean[level], ufespace, level);
+            if (count > 1.1)
+            {
+                Visualize(m2[level], ufespace, 10 + level);
+            }
         }
     }
     if (save_statistics)
     {
-        SaveFigure(mean_upscaled, ufespace, "coarse_mean");
-        SaveFigure(mean_fine, ufespace, "fine_mean");
-        SaveFigure(m2_upscaled, ufespace, "coarse_variance");
-        SaveFigure(m2_fine, ufespace, "fine_variance");
+        SaveFigure(mean[1], ufespace, "coarse_mean");
+        SaveFigure(mean[0], ufespace, "fine_mean");
+        SaveFigure(m2[1], ufespace, "coarse_variance");
+        SaveFigure(m2[0], ufespace, "fine_variance");
```

Comment:
I am not sure the purpose of saving these figures, but we may want to save all the levels here?